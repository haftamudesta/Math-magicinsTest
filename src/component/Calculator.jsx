import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.eventHandlerCalc = this.eventHandlerCalc.bind(this);
  }

  eventHandlerCalc(e) {
    this.setState((state) => calculate(state, e.target.textContent));
  }

  render() {
    const { total, operation, next } = this.state;

    return (
      <div className="calculatorContainer">
        <div className="resultScreen">
          {(total || '')
            + (operation || '')
            + (next || '')}
        </div>
        {' '}
        {/* eslint-disable-line */}
        <div className="calculatorButtons">
          <button type="button" onClick={this.eventHandlerCalc}>
            AC
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            +/-
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            %
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            รท
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            7
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            8
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            9
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            x
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            4
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            5
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            6
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            -
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            1
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            2
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            3
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            +
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            0
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            .
          </button>
          <button type="button" onClick={this.eventHandlerCalc}>
            =
          </button>
        </div>
      </div>
    );
  }
}

export default Calculator;
