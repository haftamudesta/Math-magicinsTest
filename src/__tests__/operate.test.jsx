import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import operate from '../logic/operate';
import Calculator from '../component/Calculator';
import calculate from '../logic/calculate';

test('Should render the calculate component', () => {
  const calculate = renderer.create(<calculate />).toJSON();
  expect(calculate).toMatchSnapshot();
});

test('AC should be on the document', () => {
  render(<Calculator />);
  const calculatorElement = screen.getByText('AC');
  expect(calculatorElement).toBeInTheDocument();
});

describe('Check if operate function gives the results', () => {
  const obj = {
    a: 200,
    b: 50,
  };
  test('Test multiply a * b to get 10,000', () => {
    expect(operate(obj.a, obj.b, 'x')).toBe('10000');
  });

  test('Test subtract a - b to get 150', () => {
    expect(operate(obj.a, obj.b, '-')).toBe('150');
  });

  test('Test Add a + b to get 250', () => {
    expect(operate(obj.a, obj.b, '+')).toBe('250');
  });

  test('Test divide a รท b to get 4', () => {
    expect(operate(obj.a, obj.b, 'รท')).toEqual('4');
  });

  test('Test remainder a % b to get 0', () => {
    expect(operate(obj.a, obj.b, '%')).toEqual('0');
  });
});

describe('Testing calculate', () => {
  const obj = {
    total: null,
    next: null,
    operation: null,
  };
  it('AC to be null', () => {
    obj.next = 88;
    const result = calculate(obj, 'AC');
    expect(result).toMatchObject({ total: null, next: null, operation: null });
  });
  it('AC to be null', () => {
    obj.next = 88;
    const result = calculate(obj, 'AC');
    expect(result).toMatchObject({ total: null, next: null, operation: null });
  });
  it('= is pressed with all the fields filled, it should return total with the operation executed', () => {
    obj.next = '12';
    obj.operation = '+';
    obj.total = '88';
    const result = calculate(obj, '=');

    expect(result.total).toBe('100');
  });
  it('3 is pressed with next already have 12, it should finih as 123', () => {
    obj.next = '12';
    obj.operation = null;
    obj.total = null;
    const result = calculate(obj, '3');

    expect(result.next).toBe('123');
  });
});
