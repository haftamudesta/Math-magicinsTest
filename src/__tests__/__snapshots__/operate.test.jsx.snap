// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the calculate component 1`] = `<calculate />`;
